Update documentation on TAS+ProvReq (#5417)
kubernetes-sigs/kueue: site/content/en/docs/concepts/topology_aware_scheduling.md

{{< include "examples/tas/sample-job-preferred.yaml" "yaml" >}}

### ClusterAutoscaler support

TAS integrates with the [Kubernetes ClusterAutoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler)
through the [Provisioning AdmissionCheck](/docs/admission-check-controllers/provisioning/).

> When a workload is assigned to the TAS Resource Flavor with Provisioning

Reviewer: ```suggestion
When a workload is assigned to the TAS ResourceFlavor with Provisioning
```

The `ResourceFlavor` is object name, I think.

AdmissionCheck, then its admission flow has the following stages:

> 1. **Quota reservation**: quota is reserved, and the workload obtains the

Reviewer: ```suggestion
1. **Quota reservation**: quota is reserved, and the Workload obtains the
```

In this description, it looks like `workload` indicates `Workload` object instead of generic term.

  `QuotaReserved` condtion. Preemptions are evaluated if configured.

> 2. **Admission checks**: waiting for all AdmissionChecks, including the

Reviewer: ```suggestion
2. **Admission checks**: Kueue waits for all AdmissionChecks, including the
```

>   Provisioning one, to report `Ready`.

Reviewer: ```suggestion
  Provisioning one. Once all AdmissionChecks are ready, the Workload gets `Ready` admissionCheck state.
```

Does this `Ready` indicate for `AdmissionCheck` or `Workload`?

Author: it is under `workload.status.admissionChecks[].state`

Author: >  Once all AdmissionChecks are ready, the Workload gets `Ready` admissionCheck state.

This is not exactly accurate I think, because the fields `workload.status.admissionChecks[].state` are updated by AC controllers one by one

Reviewer: ```
2. **Admission checks**: The Workloads wait for all AdmissionChecks Ready, including the
  Provisioning one.
```

In that case, how about this one?

> 3. **Topology assignment**: TAS, in the second pass of the Kueue scheduler,
>   calculates available capacity, including any newly provisioned nodes, and
>   updates the assignment with the assignment of topology domains. Once done the

Reviewer: ```suggestion
3. **Topology assignment**: Once TAS updates the topology assignments, the
```

Shouldn't we avoid TAS description here?

  `Admitted` condition is set on the Workload.

Check also [PodSet updates in ProvisioningRequestConfig](site/content/en/docs/admission-check-controllers/provisioning.md)
to see how you can configure Kueue if you want to restrict scheduling to the
newly provisioned nodes (assuming the provisioning class supports it).

### Limitations

Currently, there are limitations for the compatibility of TAS with other
features, including:
- some scheduling directives (e.g. pod affinities and anti-affinities) are ignored,
- the "podset-required-topology" annotation may fail if the underlying
  ClusterAutoscaler cannot provision nodes that satisfy the domain constraint,

> - a ClusterQueue referencing a TAS Resource Flavor (with the `.spec.topologyName`

Reviewer: ```suggestion
- a ClusterQueue references ResourceFlavors with Topology name (`.spec.topologyName`)
```

> - a ClusterQueue referencing a TAS Resource Flavor (with the `.spec.topologyName`
> field) is marked as inactive if also using [MultiKueue](multikueue.md).

Reviewer: ```suggestion
- a ClusterQueue for [MultiKueue](multikueue.md) references ResourceFlavors with Topology name (`.spec.topologyName`) is marked as inactive.
```

Does this mean that CQ for MultiKueue does not support TAS, right?

Author: Correct, currently no. We have an issue to support it: https://github.com/kubernetes-sigs/kueue/issues/5313

Author: Done
